gradle/lib/dependabot/gradle/update_checker/version_finder.rb

```diff
@@ -333,6 +330,14 @@ def central_repo_urls
         def version_class
           Gradle::Version
         end
+
+        def auth_headers_finder
+          @auth_headers_finder ||= Utils::AuthHeadersFinder.new(credentials)
+        end
+
+        def auth_headers(maven_repo_url)
+          auth_headers_finder.auth_headers(maven_repo_url)
```

Comment:
Can we memoize the result of the finder instead of the finder itself? I'm assuming it doesn't change for the lifespan of this object?

Reply:
In this case, the list of maven_repositories will be iterated and mapped, so I assume (with my minimal Ruby experience) that the method will be called multiple times with varying maven_repo_urls, why it cannot be memoized. Let me know if I am wrong. In case of the metadata_finder.rb it should be possible.

Reply:
I was wrong 😑 The change have been made.